src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -193,30 +199,44 @@ public boolean isUpdateSuccessful() {
    */
   private CompletableFuture<Void> createPieces(CompositePoLine compPOL, List<String> itemIds) {
 		CompletableFuture<Void> future = new VertxCompletableFuture<>(ctx);
-		if (itemIds.isEmpty()) {
-			future.complete(null);
-			return future;
-		}
 		List<CompletableFuture<Void>> futuresList = new ArrayList<>();
 		String poLineId = compPOL.getId();
 		String endpoint = String.format(PIECES_ENDPOINT, poLineId);
-
+		
 		handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
       .thenAccept(body -> {
         PieceCollection pieces = body.mapTo(PieceCollection.class);
-
-        // Extract item Id's which already associated with piece records
-        List<String> existingItemIds = pieces.getPieces()
-            .stream()
-            .map(Piece::getItemId)
-            .filter(StringUtils::isNotEmpty)
-            .collect(Collectors.toList());
-
-        // Create piece records for item id's which do not have piece records yet
-        itemIds.stream()
-               .filter(id -> !existingItemIds.contains(id))
-               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
-
+        int remainingPiecesToCreate = 0;
+        
+        if(itemIds!=null) {
+          // 1. Get list of item Id's which are already associated with piece records
+          List<String> existingItemIds = pieces.getPieces()
+																	             .stream()
+																	             .map(Piece::getItemId)
+																	             .filter(StringUtils::isNotEmpty)
+																	             .collect(Collectors.toList());
+
+          // 2. Get list of item id's which do not have piece records yet
+          List<String> piecesForPhysicalResources = itemIds.stream()
+          .filter(id -> !existingItemIds.contains(id)).collect(Collectors.toList());
+
+          // 3. Create piece records     
+          piecesForPhysicalResources.stream().forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));  
+	        
+	        // 4. Calculate count for remaining pieces to create when Physical and Electronic resources exists
+	        // Scenario: PO Line of "P/E Mix" format with 3 Physical resources and 2 Electronic resources with `create_inventory` set to `false`.
+	        // Create 2 piece records for Electronic resources
+	        remainingPiecesToCreate = Math.abs(calculateTotalQuantity(compPOL.getCost()) - piecesForPhysicalResources.size());   
```

Comment:
The scenario of failed electronic resources in case of P/E is missing here. Say there are 2 physical and 4 electronic to be created. In the first run 2 physical and 2 electronic are created and then call fails. In the next try, remainingPiecesToCreate would still be 4, whereas we should be creating pieces for just the remaining 2 electronic pieces.

Reply:
I have updated code so I think this scenario will be handled as well. Thanks